ASM/c/z64.h

```diff
@@ -607,16 +630,29 @@ typedef struct
                                             /* 0x1450 */
 } z64_file_t;
 
+typedef struct {
+  uint8_t *readBuff;  /* 0x00 */
+} SramContext; // size = 0x4
+
+typedef struct {
+  uint8_t data[0xBA8];
```

Comment:
So the entirety of the extra save data is one single array of bytes? I am curious if this can be more explicitly defined so that it is more obvious how many byes are available and what is where.

Reply:
0xBA8 is the extra space that is freed up in an SRAM slot. z64_file_t is the normal slot, something like 0x1450 bytes i think? Adding the 0xBA8 that makes each slot 0x1FF8 instead of the normal 0x1450. This struct was only actually modified because it was screwing w/ the file menu drawing.

The amount of that which is used I actually don't know off hand. It's dynamically calculated when the rom is patched. The new flags use a dynamic amount of space based on what settings are enabled, growing from the back of the extra SRAM slot. The remaining space at the end of 0x1450 is currently unused, but there are plans to use it for things like silver rupee shuffle.  See notes/SRAM.txt and flag_system.txt which explain how the data is structured.

Hope that explains it?

Reply:
Don't drink and code ;P